File: backend/app/schemas/owner.py
from datetime import datetime

from pydantic import BaseModel, Field


class OwnerPlaceSummary(BaseModel):
    id: str
    name: str
    address: str


class OwnerDashboardStats(BaseModel):
    total_visits: int
    this_month_visits: int
    upcoming_count: int


class OwnerMeetingVisit(BaseModel):
    meeting_id: str
    name: str
    category: str
    member_count: int
    scheduled_at: datetime
    is_upcoming: bool
    place_signal_count: int


class OwnerDashboardResponse(BaseModel):
    place: OwnerPlaceSummary
    stats: OwnerDashboardStats
    meetings: list[OwnerMeetingVisit]


class RecommendedAction(BaseModel):
    type: str
    label: str
    template: str


class OwnerCategoryInsight(BaseModel):
    category: str
    count: int
    percentage: int
    avg_member_count: int
    recommended_action: RecommendedAction


class OwnerInsightsResponse(BaseModel):
    total_meetings: int
    top_categories: list[OwnerCategoryInsight]


class OwnerMessageResponse(BaseModel):
    message: str
    active: bool


class OwnerMessageUpdateRequest(BaseModel):
    message: str = Field(default="", max_length=100)
    active: bool = False


class TimeslotInsight(BaseModel):
    key: str
    label: str
    count: int
    percentage: int


class OwnerTimeslotInsightsResponse(BaseModel):
    total_events: int
    slots: list[TimeslotInsight]
    peak_slot: str | None = None
    low_slot: str | None = None
    peak_recommendation: str = ""


class OwnerRecommendationTargetsResponse(BaseModel):
    categories: list[str]


class OwnerRecommendationTargetsUpdateRequest(BaseModel):
    categories: list[str] = Field(default_factory=list)

File: frontend/lib/types/owner.ts
export type OwnerTabId = "home" | "insights" | "tools";

export type OwnerRankingSummary = {
  district: string;
  category: string;
  rank: number | null;
  score: number;
  listed_count: number;
};

export type OwnerDashboard = {
  place: {
    id: string;
    name: string;
    address: string;
    district: string;
    category: string;
  };
  stats: {
    total_visits: number;
    this_month_visits: number;
    upcoming_count: number;
  };
  meetings: OwnerMeetingVisit[];
  ranking: OwnerRankingSummary | null;
};

export type OwnerMeetingVisit = {
  meeting_id: string;
  name: string;
  category: string;
  member_count: number;
  scheduled_at: string;
  is_upcoming: boolean;
  place_signal_count: number;
};

export type OwnerMessage = { message: string; active: boolean };

export type RecommendedAction = {
  type: string;
  label: string;
  template: string;
};

export type CategoryInsight = {
  category: string;
  count: number;
  percentage: number;
  avg_member_count: number;
  recommended_action: RecommendedAction;
};

export type OwnerInsights = {
  total_meetings: number;
  top_categories: CategoryInsight[];
};

export type TimeslotInsight = {
  key: string;
  label: string;
  count: number;
  percentage: number;
};

export type OwnerTimeslotInsights = {
  total_events: number;
  slots: TimeslotInsight[];
  peak_slot: string | null;
  low_slot: string | null;
  peak_recommendation: string;
};
